File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(number)
    sounds = {3 => 'Pling', 5 => 'Plang', 7 => 'Plong'}
    sound = sounds.map {|divisor, effect| effect if number % divisor == 0}.join
    sound.empty? ? number.to_s : sound
    # if sound.empty? then number.to_s else sound end
  end

end

File: python/raindrops/raindrops.py
import code


def convert(number):
    global sound
    sound = ""

    pling(number)
    plang(number)
    plong(number)
    if sound == "":
        return f"{number}"
    else:
        return sound


def pling(number):
    if number % 3 == 0:
        global sound
        sound += 'Pling'


def plang(number):
    if number % 5 == 0:
        global sound
        sound += 'Plang'


def plong(number):
    if number % 7 == 0:
        global sound
        sound += 'Plong'
